mmpose/core/post_processing/matching.py

```diff
@@ -0,0 +1,289 @@
+        for k in range(num_joints):
+            assert len(person[k]) > 0
+            x, y, s = person[k][0]
+            final_pose[k][0] = x.item()
+            final_pose[k][1] = y.item()
+            final_pose[k][2] = s.item()
+            if s.item() > in_vis_thre:
+                mean_score += s.item()
+                vaild_num += 1
+
+        if torch.max(final_pose[:, 2]).item() < 0.05:
+            continue
+        if vaild_num != 0:
+            mean_score = mean_score / vaild_num
+
+        x1, y1, x2, y2 = person['bbox']
+        final_result.append({
+            'keypoints':
+            final_pose.numpy(),
+            'center':
+            np.array([((x1 + x2) / 2).item(), ((y1 + y2) / 2).item()]),
+            'scale':
+            np.array([(x2 - x1).item(), (y2 - y1).item()]),
+            'area': (x2 - x1).item() * (y2 - y1).item(),
+            'score':
+            person['bbox_score'].item() * mean_score,
+            'image_id':
+            img
+        })
+
+    return final_result
+
+
+def _grouping(bboxes, pose_preds, num_joints, box_scores):
+    """remove the joints that are repeated with group.
+
+    Note:
+        num_person: N
+        num_keypoints: K
+    Args:
+        bboxes (torch.Tensor[n,4]): bbox(x1,y1,x2,y2).
+        pose_preds (dict): candidate keypoints information.
+            pose_preds[i][j] (list): i-th person j-th joint 's canditate
+                    keypoints information.
+        num_joints (int): num_keypoints.
+        box_scores (torch.Tensor[n,1]): bbox score.
+    Returns:
+        kp_groups (dict[k,dict]): information after grouping.
+    """
+
+    kp_groups = {k: {} for k in range(num_joints)}
+
+    ids = np.zeros(num_joints)
+
+    for n, person in pose_preds.items():
+        pose_preds[n]['bbox'] = bboxes[n]
+        pose_preds[n]['bbox_score'] = box_scores[n]
+        pose_preds[n]['group_id'] = {}
+        s = 0
+        for k in range(num_joints):
+            pose_preds[n]['group_id'][k] = None
+            pose_preds[n][k] = np.array(pose_preds[n][k])
+            assert len(pose_preds[n][k]) > 0
+            s += pose_preds[n][k][0][-1]
+
+        s /= num_joints
+
+        pose_preds[n]['person_score'] = s
+
+        for k in range(num_joints):
+            latest_id = ids[k]
+            kp_group = kp_groups[k]
+
+            assert len(person[k]) > 0
+            x0, y0, s0 = person[k][0]
+            if s0 < 0.05:
+                continue
+            for g_id, g in kp_group.items():
+
+                x_c, y_c = g['group_center']
+                # Get Average Box Size
+                group_area = g['group_area']
+                group_area = group_area[0] * group_area[1] / (group_area[2]**2)
+
+                # Grouping Criterion
+
+                # Joint Group
+                dist = np.sqrt(((x_c - x0)**2 + (y_c - y0)**2) / group_area)
+
+                if dist <= 0.1 * sigmas[k]:  # Small Distance
+                    if s0 > 0.3:
```

Comment:
Lots of super-parameters here. Shall we add some args to this function?

score_thr,   dist_thr  ...